transformers4rec/torch/model/base.py

```diff
@@ -678,6 +698,136 @@ def _get_name(self):
 
         return super(Model, self)._get_name()
 
+    @property
+    def input_schema(self):
+        # return the input schema given by the model
+        # loop over the heads to get input schemas
+        schemas = []
+        for head in self.heads:
+            schemas.append(head.body.inputs.schema)
+        model_schema = sum(schemas, Schema())
+
+        # TODO: rework T4R to use Merlin Schemas.
+        # In the meantime, we convert model_schema to merlin core schema
+        core_schema = Core_Schema()
+        for column in model_schema:
+            name = column.name
+
+            dtype = {0: np.float32, 2: np.int64, 3: np.float32}[column.type]
+            tags = column.tags
+            is_list = column.value_count.max > 0
+            if is_list:
+                # T4Rec is only supporting padded dense input tensors with a
+                # fixed max_sequence_length
+                max_sequence_length = column.value_count.max
+            else:
+                max_sequence_length = 1
+
+            value_count = {"min": max_sequence_length, "max": max_sequence_length}
+            is_ragged = is_list and value_count.get("min", 0) != value_count.get("max", 0)
+            int_domain = {"min": column.int_domain.min, "max": column.int_domain.max}
+            batch_dim = [-1]
+            column_dim = [max_sequence_length] if is_list else [1]
+            properties = {
+                "value_count": value_count,
```

Comment:
we can drop this value count, since for serving got it working with variable sequence inputs.